lib/rust/performance-logger/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+//!The performance-logger library allows to create performance reports via the
+//! [Web Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance).
+//!
+//! The API provided allows the marking of intervals between a start and end, which the gets
+//! measured and marked in the Chrome DevTools Performance Monitor. Intervals can be assigned a
+//! log level, for example, `TASK`, SECTION` or `DEBUG` to allow filtering based on the current
+//! needs.
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+
+use enso_prelude::*;
+
+use enso_prelude::fmt::Formatter;
+use enso_web::performance;
+use ordered_float::OrderedFloat;
+use serde::Deserialize;
+use serde::Serialize;
+use web_sys::PerformanceEntry;
+
+
+
+// ========================================
+// === Compile time filtering constants ===
+// ========================================
+
+macro_rules! define_profiling_toggle {
```

Comment:
No docs – its hard to parse macros when reading it, it should describe what it does with an example output in the doc.

Reply:
Added docs.